examples/PaletteApplicationExample.py

```diff
@@ -0,0 +1,162 @@
+        self.pal_2 = pg.palette.get('relaxed_dark')
+        self.pal_3 = pg.palette.get('relaxed_light')
+        self.mpal_1 = pg.palette.make_monochrome('green')
+        self.mpal_2 = pg.palette.make_monochrome('amber')
+        self.mpal_3 = pg.palette.make_monochrome('blue')
+
+        self.lastTime = time()
+        self.fps = None
+        self.timer = QtCore.QTimer(singleShot=False)
+        self.timer.timeout.connect( self.timed_update )
+        
+        self.timed_update()
+    
+    def testSignal(self, val):
+        """ demonstrate use of PaletteChanged signal """
+        print('"Palette changed" signal was received with value', val)
+        
+
+    def handle_button_timer_on(self):
+        """ (re-)activate timer """
+        self.timer.start(1)
+
+    def handle_button_timer_off(self):
+        """ de-activate timer """
+        self.timer.stop()
+
+    def handle_button_pal1(self):
+        """ apply palette 1 on request """
+        self.pal_1.apply()
+
+    def handle_button_pal2(self):
+        """ apply palette 2 on request """
+        self.pal_2.apply()
+
+    def handle_button_pal3(self):
+        """ apply palette 1 on request """
+        self.pal_3.apply()
+
+    def handle_button_mono1(self):
+        """ apply monochrome palette 1 on request """
+        self.mpal_1.apply()
+
+    def handle_button_mono2(self):
+        """ apply monochrome palette 2 on request """
+        self.mpal_2.apply()
+
+    def handle_button_mono3(self):
+        """ apply monochrome palette 3 on request """
+        self.mpal_3.apply()
+
+    def handle_button_leg1(self):
+        """ test legacy background / foreground overrides """
+        pg.setConfigOption('background', '#ff0000')
+        pg.setConfigOption('foreground', '#0000ff')
+
+    def handle_button_leg2(self):
+        """ test legacy background / foreground overrides """
+        pg.setConfigOption('background', '#0000ff')
+        pg.setConfigOption('foreground', '#ff0000')
+
+    def timed_update(self):
+        """ update loop, called by timer """
+        self.data1[:-1] = self.data1[1:]
+        self.data1[-1] = +3 + np.random.normal()
+        xdata = np.arange( len(self.data1) )
+        self.curve1.setData( x=xdata, y=self.data1 )
+
+        self.data2[:-1] = self.data2[1:]
+        self.data2[-1] = -3 + np.random.normal()
+        self.curve2.setData( y=self.data2 )
+
+        now = time()
+        dt = now - self.lastTime
+        self.lastTime = now
+        if self.fps is None:
+            self.fps = 1.0/dt
+        else:
+            s = np.clip(dt*3., 0, 1)
+            self.fps = self.fps * (1-s) + (1.0/dt) * s
+        self.plt.setTitle('%0.2f fps' % self.fps)
+        QtWidgets.QApplication.processEvents()  ## force complete redraw for every plot
+
+mkQApp("Palette test application")
+main_window = MainWindow()
+
+## Start Qt event loop
+if __name__ == '__main__':
+    QtWidgets.QApplication.instance().exec_()
```

Comment:
I think we updated these since you created this PR, but put this line here instead: `pg.mkQApp().exec_()` (this handles the case there is no `QApplication` instance, which I know you declare earlier, but makes it easier)